### PR TITLE
Add per request limits to http client

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Comment by wendigo** [comment]
Needed to have a single shared http client instance for all internal communication

File: http-client/src/main/java/io/airlift/http/client/Request.java [hunk 1]
```diff
@@ -221,6 +246,8 @@ public static Builder fromRequest(Request request)
         private Optional<HttpVersion> version = Optional.empty();
         private boolean followRedirects = true;
         private boolean preserveAuthorizationOnRedirect;
+        private Duration requestTimeout = Duration.ZERO;
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/Request.java:262]: It seems better to make these `Optional`. We actually allow zero for both of these in `HttpClientConfig` (though I'm not sure what that means in the context of a request timeout).

File: http-client/src/main/java/io/airlift/http/client/Request.java [hunk 2]
```diff
@@ -316,6 +330,18 @@ public Builder setIdleTimeout(Duration timeout)
             return this;
         }
 
+        public Builder setMaxContentLength(Optional<DataSize> maxContentLength)
+        {
+            this.maxContentLength = maxContentLength;
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/Request.java:335]: `requireNonNull`

File: http-client/src/main/java/io/airlift/http/client/Request.java [hunk 3]
```diff
@@ -316,6 +330,18 @@ public Builder setIdleTimeout(Duration timeout)
             return this;
         }
 
+        public Builder setMaxContentLength(Optional<DataSize> maxContentLength)
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/Request.java:333]: Do we need this method? We don't have it for `setVersion()`. It's fine if we need it.

**Review by electrum — APPROVED** [review]

**Pull request merged**